controllers/securitypolicy_controller.go

```diff
@@ -446,8 +449,24 @@ func (r *SecurityPolicyReconciler) updateStatusOfLinkedAPIs(ctx context.Context,
 		Namespace: &namespace, Name: policy.Name,
 	}
 
+	oldLinks := map[string]bool{}
+	newLinks := map[string]bool{}
+
+	for _, t := range policy.Status.LinkedAPIs {
+		oldLinks[t.String()] = true
+	}
+
+	for _, t := range policy.Spec.AccessRightsArray {
+		name := types.NamespacedName{Name: t.Name, Namespace: t.Namespace}
+		newLinks[name.String()] = true
+	}
+
 	// Remove links from api definitions
 	for _, t := range policy.Status.LinkedAPIs {
+		if _, ok := newLinks[t.String()]; ok {
```

Comment:
Can you please confirm my understanding? We are skipping ApiDefinition status update - removing Policy link - if ApiDefinition is currently referred by the Policy because we'll add this link again in the latter code piece since it is referred in `.spec`? Am I right? So that we don't update the Status unnecessarily

Reply:
Right